### DISCUSSION TITLE
Disable the connection showing up in vscode folder?

### DISCUSSION THREAD

**Discussion by BennyAlex** [mtxr/vscode-sqltools#1327] [Q&A] [answered]
Is it possible to not store the connection in settings.json inside .vscode folder?

In my opinion this should not be on by default if the connection is done via connectionstring.
This is a big security risk, as this file is normally commited to git and it contains the password in plain text!

**Comment by gjsjohnmurray** [comment, marked as answer]
Yes it is. You can put your connection definitions into the user-level `settings.json` file instead of the folder-level one that lives in the `.vscode` folder. Connections defined at user level are available to you regardless of which folder or xxx.code-workspace file you have open. They are even usable when no folder/workspace is open.

When you use the connection wizard to add a connection, the JSON will be added to one of three places:

- If no folder or workspace is open it will be added yo the user-level settings.json
- If a multi-root (or single-root) workspace is open from an XXX.code-workspace file it will be added to the `settings` object in that file.
- If a folder is open it will be added to the `.vscode/settings.json` file under that folder.

Re plaintext passwords, some SQLTools driver extensions now leverage a feature of the main SQLTools extension which helps drivers store passwords in the user's keychain on the workstation. I recommend doing this wherever available.

**Comment by BennyAlex** [comment]
Hey thanks for clearing.

I can't connect to a planetscale DB using password, only with the
connection string.

I am seeing SSL errors when using password.

Is there a guide for planetscale?

Do I need all the certificates to when enabling SSL? I thought it would
work out of the box.

John Murray ***@***.***> schrieb am Mi., 1. Mai 2024, 11:08:
